editor/modes/visual-editor/block.js

```diff
@@ -148,6 +167,8 @@ class VisualEditorBlock extends wp.element.Component {
 				onMouseMove={ this.maybeHover }
 				onMouseLeave={ onMouseLeave }
 				className={ className }
+				data-type={ block.blockType }
+				{ ...getBlockAttributesAsProps( block.attributes ) }
```

Comment:
>In all cases, blocks should be aware of the existence of a wrapper div, thus, I think we should move the `getBlockAttributesAsProps` function to the block API. Each block could decide to add any attribute to its parent div.

Yeah, I considered that, and it makes more sense now that I reflect on how the styles already make it clear that the block must be aware of the wrapper element. Your suggestion also addresses worries of computing and assigning attributes unnecessarily, dealing with complex attribute types (more accurately, not dealing with), and lends more flexibility to the block to define different / additional props to apply.